content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+
+1. variable names are a bit messy, and include different separators, spaces, and
+  irregular capitalisation
+2. variable types are partly wrong: 
+    - *unique keys* should be `character` (i.e. character strings)
+    - *dates of onset* should be `Date` (**R** is good at handling actual dates)
+    - *illness* and *sex* should be `factor` (i.e. a categorical variables)
+3. labels used in some variables are ambiguous:
+    - *sex* should be coded explicitely, not as 0 (here, male) and 1 (here,
+      female)
+	- *illness* should be coded explicitely, not as 0 (here, non-case) and 1 (case)
+4. some binary variables have maximum values of 9 (see `summary(stegen)`)
+
+
+## Data cleaning
+
+While it is tempting to go back to the Excel spreadsheet to fix issues with
+data, it is almost always quicker and more reliable to clean data in **R**
+directly. Here, we make a copy of the old data set, and clean `stegen` before
+further analysis.
+
+```{r stegen_old}
+stegen_old <- stegen # save 'dirty data'
+```
+
+We use *epitrix*'s function `clean_labels()` to standardise the variable names:
+
+```{r stegen_clean_labels}
+new_labels <- clean_labels(names(stegen)) # generate standardised labels
+new_labels # check the result
+names(stegen) <- new_labels
+```
+
+We set convert the unique identifiers to character strings (`character`), dates
+of onset to actual `Date` objects, and sex and illness are set to categorical variables (`factor`):
+
+```{r stegen_unique_ids}
+stegen$unique_key <- as.character(stegen$unique_key)
+stegen$sex <- factor(stegen$sex)
+stegen$ill <- factor(stegen$ill)
+stegen$date_onset <- as.Date(stegen$date_onset)
+```
+
+We use the function `recode()` from the *dplyr* package to recode `sex` more
+explicitely:
+
+```{r stegen_recode}
+stegen$sex <- recode_factor(stegen$sex, "0" = "male", "1" = "female")
+stegen$ill <- recode_factor(stegen$ill, "0" = "non case", "1" = "case")
+```
+
+
+Finally we look in more depth into these variables having maximum values of 9,
+where we expect 0/1; `table()` is useful to list all values taken by a variable,
+and listing their frequencies:
+
+```{r stegen_nines}
+
+table(stegen$pork)
+table(stegen$salmon)
+table(stegen$horseradish)
+
+```
+
+The only rogue values are `9`; they are likely either data entry issues, or
+missing data, which in **R** should be coded as `NA` ("not available"). We can
+replace these values using:
+
+```{r stegen_replace_9}
+stegen$pork[stegen$pork == 9] <- NA
+stegen$salmon[stegen$salmon == 9] <- NA
+stegen$horseradish[stegen$horseradish == 9] <- NA
+```
+
+
+<details>
+
+<summary> <b> Going further </b> Why does this work? Click here for more details. </summary>
+
+
+There are several things going on in a command like:
+
+```{r eval = FALSE}
+stegen$pork[stegen$pork == 9] <- NA
+```
+
+let us break them down:
+
+1. `stegen$pork` means "get the variable called `pork` in the dataset `stegen`"
+2. `[...]` subset the vector according to `...`; if `...` is a series of
```

Comment:
not sure old folk will understand "..."; realise it is the norm, but maybe replace with x and y or something?

Reply:
I had that thought as well. I think maybe replacing `...` with just `x` would serve the same purpose. It may even be better to describe it in terms of how people would do it in excel?

Reply:
I attempted to fix this. We'll see how it goes over ᕕ( ᐛ )ᕗ